Python quit unexpectedly while using the libopencv_highgui.2.4.dylib plug-in

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => 2.4.13.2
- Operating System / Platform => OS X El Capitan
- Language: Python

##### Detailed description

<!-- your description -->
OpenCv library with python client works perfectly for the image processing. However, it throws up error when I try to use VideoCapture method.

##### Steps to reproduce

<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example
    ```
    ```.py
     import numpy as np
import cv2
import sys

cap = cv2.VideoCapture("peopleCounter.avi")

w = cap.get(3) #get width
h = cap.get(4) #get height

mx = int(w/2)
my = int(h/2)

count = 0
ret, frame = cap.read()
while(cap.isOpened()):
    ret, frame = cap.read() #read a frame
    try:
        count = count + 1
        text = "Hello World " + str(count)
        cv2.putText(frame, text ,(mx,my),cv2.FONT_HERSHEY_SIMPLEX
                    ,1,(255,255,255),1,cv2.LINE_AA)
        cv2.imshow('Frame',frame)
    except:
        #if there are no more frames to show...
        print('EOF')
        break

    #Abort and exit with 'Q' or ESC
    k = cv2.waitKey(1000) & 0xff
    if k == 27:
        break

cap.release() #release video file
cv2.destroyAllWindows() #close all openCV windows
     ```
 or attach as .txt or .zip file
-->
 ```.py
import numpy as np
import cv2
import sys

cap = cv2.VideoCapture("peopleCounter.avi")

w = cap.get(3) #get width
h = cap.get(4) #get height

mx = int(w/2)
my = int(h/2)

count = 0
ret, frame = cap.read()
while(cap.isOpened()):
    ret, frame = cap.read() #read a frame
    try:
        count = count + 1
        text = "Hello World " + str(count)
        cv2.putText(frame, text ,(mx,my),cv2.FONT_HERSHEY_SIMPLEX
                    ,1,(255,255,255),1,cv2.LINE_AA)
        cv2.imshow('Frame',frame)
    except:
        #if there are no more frames to show...
        print('EOF')
        break

    #Abort and exit with 'Q' or ESC
    k = cv2.waitKey(1000) & 0xff
    if k == 27:
        break

cap.release() #release video file
cv2.destroyAllWindows() #close all openCV windows
````
##### Error messsage
<img width="1007" alt="screen shot 2017-03-02 at 3 44 18 am" src="https://cloud.githubusercontent.com/assets/15026232/23499735/4b2a807c-fefb-11e6-9854-e68f70e5031e.png">

@susanbhattarai Thanks! I wanted to make sure it wasn't because I was using some outdated library, since the C-API is more or less no longer being extended.

@alalek Your code fix worked! I compiled locally and ran my C && C++ examples ran without exception. I've included a PR. https://github.com/opencv/opencv/pull/8979

The python example I wrote played the video, but had a different error which isn't related to this fix:
```
opencv/modules/imgproc/src/color.cpp:3739: error: (-215) scn == 3 || scn == 4 in function cvtColor
```

--- a/modules/highgui/src/cap_avfoundation_mac.mm
+++ b/modules/highgui/src/cap_avfoundation_mac.mm
@@ -697,7 +697,15 @@ CvCaptureFile::CvCaptureFile(const char* filename) {
         return;
     }
 
-    mAssetTrack = [[mAsset tracksWithMediaType: AVMediaTypeVideo][0] retain];
+    NSArray *tracks = [mAsset tracksWithMediaType:AVMediaTypeAudio];
+    if ([tracks count] == 0) {
+      fprintf(stderr, "OpenCV: Couldn't read movie file \"%s\"\n", filename);
+      [localpool drain];
+      started = 0;
+      return;
+    }
+
+    mAssetTrack = [tracks[0] retain];
 
     if ( ! setupReadingAt(kCMTimeZero) ) {
         fprintf(stderr, "OpenCV: Couldn't read movie file \"%s\"\n", filename);

